hardwarelibrary/communication/sutter.py

```diff
@@ -0,0 +1,149 @@
+from struct import *
+import sys
+import serial
+from serial.tools.list_ports import comports
+#sys.path.insert(0, "/Users/justinemajor/Documents/gph.doc/git/PyHardwareLibrary/hardwarelibrary/communication")
+import time
+import array
+import os
+import serial
+
+#/dev/cu.usbserial-SI8YCLBE
+class SutterDevice:
+    def __init__(self):
+        """
+        SutterDevice represents a XYZ stage.  
+        """
+        
+        self.port = serial.Serial("/dev/cu.usbserial-SI8YCLBE", timeout=5, baudrate=128000)
+
+        #ajouter les instances nécessaires
+
+        self.microstepsPerMicrons = 16
+
+    def initializeDevice(self):
+        """
+        We do a late initialization: if the device is not present at creation, it can still be
+        initialized later.
+        """
+        if self.port is not None:
+            return
+        
+        self.port = serial.Serial("/dev/cu.usbserial-SI8YCLBE", timeout=5, baudrate=128000) 
+        if self.port is None:
+            raise IOError("Can't find Sutter device")
+
+    def shutdownDevice(self):
+        """
+        If the device fails, we shut everything down. We should probably flush the buffers also.
+        """
+        self.port.close()
+
+    def sendCommand(self, commandBytes):
+        """ The function to write a command to the endpoint. It will initialize the device 
+        if it is not alread initialized. On failure, it will warn and shutdown."""
+        try:
+            if self.port is None:
+                self.initializeDevice()
+            
+            self.port.write(commandBytes)
+
+        except Exception as err:
+            print('Error when sending command: {0}'.format(err))
+            self.shutdownDevice()
+
+    def readReply(self, size, format) -> tuple:
+        """ The function to read a reply from the endpoint. It will initialize the device 
+        if it is not already initialized. On failure, it will warn and shutdown. 
+        It will unpack the reply into a tuple.
+        """
+        try:
+            if self.port is None:
+                self.initializeDevice()
+
+            replyBytes = self.port.read(size)
+            if len(replyBytes) == size:
+                theTuple = unpack(format, replyBytes)
+                return theTuple
+
+        except Exception as err:
+            print('Error when reading reply: {0}'.format(err))
+            self.shutdownDevice()
+            return None
+
+    def positionInMicrosteps(self) -> (int,int,int):
+        """ Returns the position in microsteps """
+        commandBytes = pack('<cc', b'C', b'\r')
+        self.sendCommand(commandBytes)
+        return self.readReply(size=14, format='<clllc')[1:-1]
```

Comment:
This is not robust: first and last characters donèt need to be removed from the tuple.